packages/office-ui-fabric-react/src/components/ComboBox/ComboBox.tsx

```diff
@@ -293,7 +296,14 @@ export class ComboBox extends BaseComponent<IComboBoxProps, IComboBoxState> {
         </div>
 
         { isOpen && (
-          (onRenderContainer as any)({ ...this.props as any }, this._onRenderContainer)
+          (onRenderContainer as any)({
+            ...this.props,
+            onRenderList: onRenderList,
+            onRenderItem: onRenderItem,
+            onRenderOption: onRenderOption,
+            options: this.state.currentOptions.map((item, index) => ({ ...item, index: index }))
```

Comment:
I moved this up from the default onRenderList.
We could also assign the selected state of the item here as well, if we want to do that